Updates for gradle deprecations that are removed in Gradle 9

jhanders34 opened the pull request:
Gradle 9 specific updates
- Change from using `prop value` and `prop (value)` to doing `prop = value`
- Update springboot gradle plugin version to 3.5.3 consistently for spring boot 2 and 3 test applications since the older versions didn't work with gradle 9
- Since using spring boot 3 gradle plugin for building spring boot 2 applications, updated all spring boot 2 applications to be override spring 3 settings from the gradle plugin so that the war and jar files are like they were before with gradle 8 and using the spring boot 2 gradle plugin
- Update to bnd gradle plugin 7.0 and update compileJavaTest to add the built jar to the classpath which is a behavior change in 7.0
- Update to surround sourceCompatibility with `java { }` 
- Switch from `fileMode` to `filePermissions`
- Update import for XmlSluper
- Update from `javaexec` to `providers.javaexec`
- Change from using `distsDirectory` to using `layout.buildDirectory.dir("distributions")`
- Switch for `archivePath` to `archiveFile`
- Switch from `destinationDir` to destinationDirectory`
- Add some new dependsOn for errors when running with Gradle 9
- Add `org.gradle.archives.use-file-system-permissions=true` to gradle.properties to use the old logic for file permisisons in archives
- Remove unit test that doesn't actually have any tests in it since that isn't allowed any longer

Gradle 10 specific updates
- Started the work to not reference Task.project in doLast.  More will be done in a follow on PR
- Rename `test` directories to `test_resources` directories that do not have unit tests in the directory.  In Gradle 10 this will not be allowed

Gradle switch over
- Updated to switch to Gradle 9.1 for running locally.  Builds run on the build systems will still use Gradle 8 until they are switched over.

- [x] I have considered the risk of behavior change or other zero migration impact (https://github.com/OpenLiberty/open-liberty/wiki/Behavior-Changes).
- [x] If this PR fixes an Issue, the description includes "Fixes #FILLMEIN" or "Resolves #FILLMEIN" (verify `release bug` label if applicable: https://github.com/OpenLiberty/open-liberty/wiki/Open-Liberty-Conventions).
- [x] If this PR resolves an external Known Issue (including APARS), the description includes "Fixes #FILLMEIN" or "Resolves #FILLMEIN".

jhanders34 committed:
Updates for gradle deprecations that are removed in Gradle 9

fritze2 commented:
I requested a Release pipeline to test at https://libh-proxy1.fyre.ibm.com/cognitive-new/pipelineAnalysis.html?pipelineId=c2a6e2a2-ee8e-4290-816d-86f8c7c30997 since the !release keyword doesn't trigger in OL.

wraschke commented:
Hello, Jared! Ignore this comment.

jhanders34 committed:
Update springboot gradle plugin use to 3.5.3

- Updates spring boot 2 and 3 test applications to use the 3.5.3
version of the gradle plugin.  For spring boot 2 there are updates
to be able to build a version 2 application with a version 3 plugin

jhanders34 committed:
Update import for XmlSlurper

jhanders34 committed:
Switch from using currentProject to not referencing project at all

fritze2 commented on dev/cnf/build.gradle:
```diff
@@ -128,16 +128,17 @@ task dependabotGenerate() {
     dependsOn ':wlp-bndPlugins:jar'
     dependsOn dependabotGenerate
     from zipTree(bnd.get('build.bnd.plugins.jar'))
     into project.file('bndplugins/classes')
```
Should drop the `project.`?

jhanders34 replied:
It isn't in a doLast so it isn't a problem for Gradle 10.

fritze2 commented on dev/build.pii.package/build.gradle:
```diff
@@ -15,25 +15,25 @@ task zipMessageFiles(type: Zip) {
   publications {
     maven(MavenPublication) {
-      artifactId 'openPII'
-      version project.version
```
Does `project.` matter on the version too? There's a handful of these.

jhanders34 replied:
This isn't in a doLast so it isn't an issue.  We need to use project.version to specify what version it is.

jhanders34 committed:
Update to Gradle 9.1.0

fritze2 commented:
!build
 ([view Open Liberty Personal Build](https://libh-proxy1.fyre.ibm.com/cognitive/pipelineAnalysis.html?pipelineId=8e3697f1-9e09-4ed9-814c-d70709ba02c6) - :hourglass_flowing_sand:InProgress)
gradle.version=9.1.0

_Note: Target locations of links might be accessible only to IBM employees._

jhanders34 committed:
Add depends that is needed when running with gradle 9

jhanders34 commented:
!build
 ([view Open Liberty Personal Build](https://libh-proxy1.fyre.ibm.com/cognitive/pipelineAnalysis.html?pipelineId=4f113c6e-10a2-43e3-ade6-5840e9de246e) - :hourglass_flowing_sand:InProgress)
gradle.version=9.1.0

_Note: Target locations of links might be accessible only to IBM employees._

jhanders34 commented:
!fullbuild ([view Open Liberty Personal Build](https://libh-proxy1.fyre.ibm.com/cognitive/pipelineAnalysis.html?pipelineId=b469802d-c5fb-4f50-8a04-a1f05c542557) - :hourglass_flowing_sand:InProgress)

_Note: Target locations of links might be accessible only to IBM employees._

jhanders34 commented:
!fullbuild
 ([view Open Liberty Personal Build](https://libh-proxy1.fyre.ibm.com/cognitive/pipelineAnalysis.html?pipelineId=b8215e03-480c-4b79-a27b-a6341de124ec) - :hourglass_flowing_sand:InProgress)
gradle.version=9.1.0

_Note: Target locations of links might be accessible only to IBM employees._

jhanders34 commented:
!build
 ([view Open Liberty Personal Build](https://libh-proxy1.fyre.ibm.com/cognitive/pipelineAnalysis.html?pipelineId=fc5eb6fe-2981-4cfd-bf41-a932560b779d) - :hourglass_flowing_sand:InProgress)
gradle.version=9.1.0

_Note: Target locations of links might be accessible only to IBM employees._

jhanders34 commented:
!fullbuild
 ([view Open Liberty Personal Build](https://libh-proxy1.fyre.ibm.com/cognitive/pipelineAnalysis.html?pipelineId=4b4effce-1645-49e3-9982-3e6ce1bb8157) - :hourglass_flowing_sand:InProgress)
gradle.version=9.1.0

_Note: Target locations of links might be accessible only to IBM employees._

jhanders34 commented:
!build ([view Open Liberty Personal Build](https://libh-proxy1.fyre.ibm.com/cognitive/pipelineAnalysis.html?pipelineId=04f3b3a9-842e-4956-9976-abf772895b68) - :x: completed with errors/failures.)

_Note: Target locations of links might be accessible only to IBM employees._

jhanders34 commented:
!build
 ([view Open Liberty Personal Build](https://libh-proxy1.fyre.ibm.com/cognitive/pipelineAnalysis.html?pipelineId=8d74f778-1415-4e9d-bc6e-30c13a8cf3ae) - :hourglass_flowing_sand:InProgress)
gradle.version=9.1.0

_Note: Target locations of links might be accessible only to IBM employees._

jhanders34 commented:
!fullbuild
 ([view Open Liberty Personal Build](https://libh-proxy1.fyre.ibm.com/cognitive/pipelineAnalysis.html?pipelineId=08dff726-1c77-4bbb-8631-3464d75f8875) - :hourglass_flowing_sand:InProgress)
gradle.version=9.1.0

_Note: Target locations of links might be accessible only to IBM employees._

jhanders34 commented:
!fullbuild ([view Open Liberty Personal Build](https://libh-proxy1.fyre.ibm.com/cognitive/pipelineAnalysis.html?pipelineId=7f839fb9-29dc-40ed-992f-7f0b9f35bfe2) - :x: completed with errors/failures.)

_Note: Target locations of links might be accessible only to IBM employees._

jhanders34 committed:
Remove unit test that doesn't have any tests in it since that isn't allowed any more.

jhanders34 commented:
!build
 ([view Open Liberty Personal Build](https://libh-proxy1.fyre.ibm.com/cognitive/pipelineAnalysis.html?pipelineId=d7f3e291-3db0-45ef-b6f9-8cfdaeba10a0) - :hourglass_flowing_sand:InProgress)
gradle.version=9.1.0

_Note: Target locations of links might be accessible only to IBM employees._

jhanders34 commented:
!fullbuild
 ([view Open Liberty Personal Build](https://libh-proxy1.fyre.ibm.com/cognitive/pipelineAnalysis.html?pipelineId=dad269fc-5fb4-40e8-83dd-67bbb4edcc23) - :hourglass_flowing_sand:InProgress)
gradle.version=9.1.0

_Note: Target locations of links might be accessible only to IBM employees._

jhanders34 commented:
!fullbuild ([view Open Liberty Personal Build](https://libh-proxy1.fyre.ibm.com/cognitive/pipelineAnalysis.html?pipelineId=c685e1b9-e59a-4a84-8573-808b87b79100) - :hourglass_flowing_sand:InProgress)

_Note: Target locations of links might be accessible only to IBM employees._

jhanders34 commented:
!build
 ([view Open Liberty Personal Build](https://libh-proxy1.fyre.ibm.com/cognitive/pipelineAnalysis.html?pipelineId=b9b26d90-8536-4e72-9122-dd775988089a) - :hourglass_flowing_sand:InProgress)
gradle.version=9.1.0

_Note: Target locations of links might be accessible only to IBM employees._

jhanders34 commented:
!build ([view Open Liberty Personal Build](https://libh-proxy1.fyre.ibm.com/cognitive/pipelineAnalysis.html?pipelineId=fbcb4ddc-3976-4772-986a-df5b44993e93) - :hourglass_flowing_sand:InProgress)

_Note: Target locations of links might be accessible only to IBM employees._

jhanders34 committed:
Add missing duplicatesStrategy

jhanders34 commented:
!build ([view Open Liberty Personal Build](https://libh-proxy1.fyre.ibm.com/cognitive/pipelineAnalysis.html?pipelineId=8392a9ce-3f35-478e-a65f-19d9432c1caf) - :x: completed with errors/failures.)

_Note: Target locations of links might be accessible only to IBM employees._

jhanders34 commented:
!build
 ([view Open Liberty Personal Build](https://libh-proxy1.fyre.ibm.com/cognitive/pipelineAnalysis.html?pipelineId=e69ad5c1-e577-49dc-bbf6-26708b96f601) - :x: completed with errors/failures.)
gradle.version=9.1.0

_Note: Target locations of links might be accessible only to IBM employees._

jhanders34 commented:
!fullbuild ([view Open Liberty Personal Build](https://libh-proxy1.fyre.ibm.com/cognitive/pipelineAnalysis.html?pipelineId=8032b51e-e64f-41e6-afc7-913a520fe1de) - :x: completed with errors/failures.)

_Note: Target locations of links might be accessible only to IBM employees._

jhanders34 commented:
!fullbuild
 ([view Open Liberty Personal Build](https://libh-proxy1.fyre.ibm.com/cognitive/pipelineAnalysis.html?pipelineId=6755acd3-7cca-4f1e-a7d1-d825a9832c9f) - :x: completed with errors/failures.)
gradle.version=9.1.0

_Note: Target locations of links might be accessible only to IBM employees._

jhanders34 committed:
Switch back to version 3.2.4 for the checkpoint spring boot data app

jhanders34 commented:
!build ([view Open Liberty Personal Build](https://libh-proxy1.fyre.ibm.com/cognitive/pipelineAnalysis.html?pipelineId=6f122462-5c50-4f0c-8a5d-bcff238d2526) - :x: completed with errors/failures.)

_Note: Target locations of links might be accessible only to IBM employees._

jhanders34 commented:
!build
 ([view Open Liberty Personal Build](https://libh-proxy1.fyre.ibm.com/cognitive/pipelineAnalysis.html?pipelineId=069f9a9a-02bd-447d-bef6-9d0f2982ed8b) - :x: completed with errors/failures.)
gradle.version=9.1.0

_Note: Target locations of links might be accessible only to IBM employees._

jhanders34 commented:
!fullbuild ([view Open Liberty Personal Build](https://libh-proxy1.fyre.ibm.com/cognitive/pipelineAnalysis.html?pipelineId=7f7d8920-8512-4911-86dd-b5bfe3c2bdd4) - :x: completed with errors/failures.)

_Note: Target locations of links might be accessible only to IBM employees._

jhanders34 commented:
!fullbuild
 ([view Open Liberty Personal Build](https://libh-proxy1.fyre.ibm.com/cognitive/pipelineAnalysis.html?pipelineId=f918c4a8-2f04-4436-9efc-126bdc6e1b30) - :x: completed with errors/failures.)
gradle.version=9.1.0

_Note: Target locations of links might be accessible only to IBM employees._

fritze2 reviewed (approved):
Looks great to me.
